jcl/src/java.base/share/classes/jdk/crac/Core.java

```
@@ -0,0 +1,111 @@
+	}
+
+	/**
+	 * Invoke InternalCRIUSupport with default settings.
+	 *
+	 * @throws CheckpointException if this failed with a CheckpointException
+	 * @throws RestoreException    if this failed with a RestoreException
+	 */
+	public static void checkpointRestore() throws CheckpointException, RestoreException {
+		if (InternalCRIUSupport.isCRaCSupportEnabled()) {
+			try {
+				globalContext.checkpointJVM();
+			} catch (Throwable t) {
+				throw new CheckpointException(t);
+			}
+		} else {
+			throw new UnsupportedOperationException("CRaC support is not enabled"); //$NON-NLS-1$
+		}
+	}
+}
+
+class CRIUSupportContext<R extends Resource> extends Context<R> {
+	private InternalCRIUSupport internalCRIUSupport = new InternalCRIUSupport(
+			Paths.get(InternalCRIUSupport.getCRaCCheckpointToDir())).setLeaveRunning(false).setShellJob(true)
+			.setFileLocks(true);
+
+	@Override
+	public void register(R resource) throws Exception {
+		if (Context.debug) {
+			System.out.print("Register: " + resource); //$NON-NLS-1$
+		}
+
+		try {
+			internalCRIUSupport.registerPreCheckpointHook(() -> {
+				try {
+					resource.beforeCheckpoint(this);
+				} catch (Throwable t) {
+					throw new JVMCheckpointException(t.getMessage(), 0, t.getCause());
+				}
+			}, InternalCRIUSupport.HookMode.CONCURRENT_MODE, InternalCRIUSupport.LOWEST_USER_HOOK_PRIORITY);
+			internalCRIUSupport.registerPostRestoreHook(() -> {
+				try {
+					resource.afterRestore(this);
+				} catch (Throwable t) {
+					throw new JVMRestoreException(t.getMessage(), 0, t.getCause());
+				}
+			}, InternalCRIUSupport.HookMode.CONCURRENT_MODE, InternalCRIUSupport.LOWEST_USER_HOOK_PRIORITY);
+		} catch (JVMCheckpointException jce) {
+			throw new CheckpointException(jce);
+		} catch (JVMRestoreException jre) {
+			throw new RestoreException(jre);
+		}
+	}
+
+	@Override
+	public void beforeCheckpoint(Context<? extends Resource> resource) throws CheckpointException {
+		// not required
```

this should throw `UnsupportedOperationException`